libopenage/util/misc_test.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright 2017-2017 the openage authors. See copying.md for legal info.
+
+#include "misc.h"
+
+#include "../testing/testing.h"
+
+
+namespace openage {
+namespace util {
+namespace tests {
+
+
+void array_conversion() {
+	{
+		static const std::vector<uint8_t> test_array8{
+			0x01, 0x23, 0x45, 0x67, 0x89, 0xab, 0xcd, 0xef,
+			0xf0, 0xf1, 0xf2, 0xf3, 0xf4, 0xf5, 0xf6, 0xf7
+		};
+		// The same array but as uint64_t
+		// ASDF: Can we just assume that everything is little-endian?
```

Comment:
hehe no :) you'd need to add one case for each endianness (for runtime switch, you maybe wanna use http://stackoverflow.com/a/1001330/1348535)